weasel/compat.py

```diff
@@ -3,3 +3,8 @@
 is_windows = sys.platform.startswith("win")
 is_linux = sys.platform.startswith("linux")
 is_osx = sys.platform == "darwin"
+
+try:
+    from importlib.metadata import version  # noqa
+except ImportError:
+    from catalogue._importlib_metadata import version  # type: ignore [no-redef]  # noqa
```

Comment:
If we're going to delay this to v4, then we can drop python 3.6 and 3.7 and we don't need this compat section.

Reply:
Done, thanks!